plugins/woocommerce-admin/client/layout/controller.js

```diff
@@ -185,47 +192,73 @@ export const getPages = () => {
 			navArgs: {
 				id: 'woocommerce-add-product',
 			},
-			wpOpenMenu: 'menu-posts-product',
-			capability: 'manage_woocommerce',
 		} );
 
 		pages.push( {
-			container: AddProductPage,
-			path: '/add-product-old',
+			...productPage,
+			path: '/product/:productId',
 			breadcrumbs: [
-				[ '/add-product-old', __( 'Product', 'woocommerce' ) ],
-				__( 'Add New Product', 'woocommerce' ),
+				[ '/edit-product', __( 'Product', 'woocommerce' ) ],
+				__( 'Edit Product', 'woocommerce' ),
 			],
 			navArgs: {
-				id: 'woocommerce-add-product-old',
+				id: 'woocommerce-edit-product',
 			},
-			wpOpenMenu: 'menu-posts-product',
-			capability: 'manage_woocommerce',
 		} );
 
+		if ( window.wcAdminFeatures[ 'new-product-management-experience' ] ) {
+			pages.push( {
+				...productPage,
+				container: AddProductPage,
+				path: '/add-product-old',
+				breadcrumbs: [
+					[ '/add-product-old', __( 'Product', 'woocommerce' ) ],
+					__( 'Add New Product (Old)', 'woocommerce' ),
+				],
+				navArgs: {
+					id: 'woocommerce-add-product',
+				},
+			} );
+
+			pages.push( {
+				...productPage,
+				container: EditProductPage,
+				path: '/product-old/:productId',
+				breadcrumbs: [
+					[ '/edit-product', __( 'Product', 'woocommerce' ) ],
+					__( 'Edit Product (Old)', 'woocommerce' ),
+				],
+				navArgs: {
+					id: 'woocommerce-edit-product',
+				},
+			} );
+		}
+	} else if (
+		window.wcAdminFeatures[ 'new-product-management-experience' ]
+	) {
```

Comment:
I think we could remove this `else if` condition if we just replace the current feature flag with the other one.

Reply:
Hmm on second thought, users are able to enable it in the last released WooCommerce version, so we probably do want to keep this around for at-least one release.
We may want to add a big banner to the old UI also that it will be replace by the other feature in the next release?